pool/src/lib.rs

```diff
@@ -21,14 +22,60 @@ mod test;
 use crate::storage::*;
 
 #[allow(dead_code)] //TODO: remove after full implement validate_borrow
-#[derive(Debug, Clone, Copy)]
+#[derive(Debug, Clone)]
```

Comment:
Let's move all the structs below out of the contract to be consistent with the other types (e.g. ReserveData).